### PR TITLE
Feature/e UI 8452

### PR DESCRIPTION
**Pull request by MateiHMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-8452

### Change description ###
Respond to query function -- add hearingDate date input field

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/palette.module.ts [hunk 1]
```diff
@@ -119,21 +119,17 @@ import { PaletteService } from './palette.service';
 import { CasePaymentHistoryViewerFieldComponent } from './payment';
 import { ReadPhoneUKFieldComponent, WritePhoneUKFieldComponent } from './phone-uk';
 import {
+  QueryCaseDetailsHeaderComponent,
+  QueryCreateComponent,
+  QueryDetailsComponent,
+  QueryDetailsTableComponent,
+  QueryListComponent,
+  QueryWriteAddDocumentsComponent, QueryWriteDateInputComponent,
```
Comment (Daniel-Lam) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/palette.module.ts:127]: @MateiHMCTS - any reason not to put this on a separate line like the others?

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-write/query-write-date-input/query-write-date-input.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    await TestBed.configureTestingModule({
+      imports: [FormsModule],
+      declarations: [QueryWriteDateInputComponent, RpxTranslateMockPipe]
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(QueryWriteDateInputComponent);
+    component = fixture.componentInstance;
+    // Call the register as CVA does
```
Comment (Daniel-Lam) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-write/query-write-date-input/query-write-date-input.component.spec.ts:27]: @MateiHMCTS - what's "CVA"?

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-write/query-write-raise-query/query-write-raise-query.component.html [hunk 3]
```diff
@@ -10,98 +10,112 @@ <h1 class="govuk-heading-l govuk-!-margin-bottom-4">{{ 'Enter query details' | r
...
-                       [checked]="formGroup.get('isHearingRelated').value === false" [value]="false"
-                >
-                <label class="govuk-label govuk-radios__label" for="query-raise-isHearingRelated-no">
-                  {{ 'Is the query hearing related?' | rpxTranslate: null : 'No' }}
+        <ng-container *ngIf="formGroup.get('isHearingRelated').value">
+          <div class="govuk-!-margin-top-1">
+            <fieldset class="govuk-fieldset" role="group">
+              <legend>
+                <label class="govuk-label govuk-label--s">
+                  {{ 'Hearing Date' | rpxTranslate }}
```
Comment (Daniel-Lam) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-write/query-write-raise-query/query-write-raise-query.component.html:112]: @MateiHMCTS - just to check whether "Date" should be "date", given "Hearing" already has an initial capital

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@MateiHMCTS - just a few things to look at. Also, I'm sure somewhere we already have a GDS-style date field - maybe in XUI Common Library. I guess we can stick with what you've implemented for now, but it's worth seeing if there is already something

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-write/query-write-date-input/query-write-date-input.component.spec.ts (hunk 2 above)
Comment (MateiHMCTS) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-write/query-write-date-input/query-write-date-input.component.spec.ts:27]: Control Value Accessor

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-write/query-write-raise-query/query-write-raise-query.component.html (hunk 3 above)
Comment (MateiHMCTS) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-write/query-write-raise-query/query-write-raise-query.component.html:112]: Made the change

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/palette.module.ts (hunk 1 above)
Comment (MateiHMCTS) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/palette.module.ts:127]: IDE added it I guess; Made the change

**Review by Daniel-Lam — APPROVED** [review]

**Review by johnbenjamin-hmcts — DISMISSED** [review]

**Comment by MateiHMCTS** [comment]
Closed as this was merged with EUI-8453

**Pull request closed without merging**